### DISCUSSION TITLE
Animation Player toolbar in 3D Mode

### DISCUSSION THREAD

**Discussion by unluckychloe** [godotengine/godot-proposals#7443] [Editor]
hi! new user, sorry if this has been asked but searched and even asked around offsite and couldnt find any answer.

what's the reasoning behind the extremely helpful animation player toolbar being absent in the 3D view? it seems to me like the functionality would be the exact same, is there some deeper technical reason it's not? i'm aware most use outside programs like blender to animate, but that doesn't make it a less bizarre omission considering it's implemented into the 2D animation pipeline. if there are quirks that make it a challenge to implement that's obviously understandable, would just love to have some kinda explanation before attempting to implement it myself via plugin so i can know if i'm wasting my time. thanks!

**Comment by theraot** [comment]
I assume you mean the toolbar. It will appear in Godot 4.x when you select Skeleton3D, before animating Skeleton3D was not a feature. Arguably it could be useful for other nodes, but you don't need it, you can animate with the Animation panel, and the inspector. I suppose it could be added for usability.

**Reply by unluckychloe** [reply]
didn't realize it was supported by skeleton3D, thank you! but i would still love if it was usable more generally in 3D. it seems like a pure UI issue - i can accomplish the same goal by manually adding position+rotation+scaling keyframes on every object for every frame, but that's obviously not optimal and doesn't scale up at all. from a outside new user perspective, not even being able to add a keyframe to multiple tracks at once seems like a UI oversight more than a missing feature.
